### ISSUE TITLE
Flutter bindings

### ISSUE THREAD

**Issue by systemed** [maplibre/maplibre-native#41]
Mapbox have a set of Flutter bindings to Mapbox GL Native: https://github.com/tobrun/flutter-mapbox-gl

It began life as an official Mapbox repo. It's now "community driven and is not an official Mapbox product... we welcome feedback and contributions". Mapbox devs still appear to be active in the project: there's a project status discussion at https://github.com/tobrun/flutter-mapbox-gl/issues/517

It would be good to have Flutter bindings for MapLibre. The Mapbox bindings are still BSD-licensed, so it should be possible to base them off the latest version, obviously without bindings to any post-licence change MBGL features.

Putting this here as a placeholder and gathering point for now - I may look at this when I need the bindings but obviously won't be disappointed if someone else gets there before me!

**Comment by m0nac0** [comment]
I've actually been working on this in https://github.com/m0nac0/flutter-maplibre-gl! 
(I'm also a contributor to https://github.com/tobrun/flutter-mapbox-gl)

Current status is: 
- Android uses Maplibre GL Native and seems to work fine, but I haven't tested features provided by the Android plugins. 
- iOS compiles but still uses Mapbox GL Native. I likely won't be able to complete the migration for iOS, since I don't have a Mac. **Help is greatly appreciated!**
- web uses Maplibre GL JS and seems to work fine.

Apart from the library migrations I've also done a lot of refactoring. 
The only breaking API changes so far are: `MapboxMap` <--> `MaplibreMap` and `MapboxMapController` <--> `MaplibreMapController`, so it should be easy to switch between the libraries.

**Comment by jwinarske** [comment]
Any runtime metrics for Android or iOS?  Such as threads, worst case reminder time, memory footprint, disk usage?  I'm interested in Linux, and may be able to provide implementation

**Comment by m0nac0** [comment]
I don't have any runtime metrics, unfortunately. Linux support would be amazing, though!

**Comment by m0nac0** [comment]
@systemed Update: [https://github.com/m0nac0/flutter-maplibre-gl](https://github.com/m0nac0/flutter-maplibre-gl) now supports Android, iOS and web! (Thanks to @mindthefish who got iOS to work!).

@jwinarske Did you ever happen to start working on Linux support?

**Comment by jwinarske** [comment]
I have been working with Mapbox directly.  So no update.

**Comment by jwinarske** [comment]
@wipfli It should remain open.  There are two options.  PlatformViews, or Render to Texture.  Using Flutter PlatformViews is less performant than using render to texture.

Also I am no longer working with Mapbox on this, so I am open to collaborate.  Tobrun's approach is not official Mapbox support, and is definitely a naive implementation.

**Comment by wipfli** [comment]
What is Tubrun?

**Comment by jwinarske** [comment]
tobrun -> https://github.com/tobrun/flutter-mapbox-gl
An employee of Mapbox, only it's not related to Mapbox deliverables.  Personal project.

**Comment by wipfli** [comment]
Thanks :) Redirects to https://github.com/flutter-mapbox-gl/maps

**Comment by birkskyum** [comment]
@jwinarske, I'm not sure I understand your answer to why to keep the issue open. Is the slow approach used in flutter-maplibre-gl, or can you rephrase it somehow?

**Comment by jwinarske** [comment]
@birkskyum

flutter-maplibre-gl:
* limited platform support (eg no generic Linux support) 
* uses platform views; not compatible for most custom embedders
* Rendering to texture is a better solution

**Comment by louwers** [comment]
There are now various libraries for MapLibre (Native) on Flutter. 

If someone wants to address the limitations of [flutter-maplibre-gl](https://github.com/maplibre/flutter-maplibre-gl) with another approach, please open a more actionable issue or PR (such as https://github.com/maplibre/maplibre-native/pull/1254).
